compiler/src/codegen/transl_anf.re

```diff
@@ -1228,44 +1235,78 @@ let lift_imports = (env, imports) => {
   (imports, setups, env);
 };
 
-let transl_signature = (functions, signature) => {
+let transl_signature = (functions, imports, signature) => {
   open Types;
 
+  let function_exports = ref([]);
+
   // At this point in compilation, we know which functions can be called
   // directly/indirectly at the wasm level. We add this information to the
   // module signature.
   let func_map = Ident_tbl.create(30);
   List.iter(
-    func => Ident_tbl.add(func_map, (func: mash_function).id, func),
+    (func: mash_function) =>
+      switch (func.name) {
+      | Some(name) =>
+        Ident_tbl.add(func_map, func.id, Ident.unique_name(func.id))
+      | None => ()
+      },
     functions,
   );
+  List.iter(
+    imp =>
+      switch (imp.imp_shape) {
+      | FunctionShape(_) =>
+        let internal_name =
+          switch (imp.imp_desc) {
+          | GrainValue(mod_, name) => grain_import_name(mod_, name)
+          | WasmFunction(mod_, name) => Ident.unique_name(imp.imp_use_id)
+          | _ => failwith("Wasm or js value had FunctionShape")
+          };
+        Ident_tbl.add(func_map, imp.imp_use_id, internal_name);
+      | _ => ()
+      },
+    imports,
+  );
   let sign =
     List.map(
       fun
       | TSigValue(
-          _,
+          vid,
           {
             val_repr: ReprFunction(args, rets, _),
             val_fullpath: Path.PIdent(id),
           } as vd,
         ) => {
           switch (Ident_tbl.find_opt(func_map, id)) {
-          | Some({name: Some(name)}) =>
+          | Some(internal_name) =>
+            let external_name = Ident.name(vid);
```

Comment:
What's the significance of changing to `vid` instead of `id` here (and line 1293)?

Reply:
This way is "guaranteed" to be correct, since it's the name that's included in the module signature rather than the name that's a part of the full path (which is something internal). 1293 was really just a bug 😶 

I guess it wasn't _really_ a bug since it doesn't affect anything, but `id` could potentially be incorrect.